File: trunk/client/python/testcli.py

import sphivedbcli
import time
import sys

def printResultSet( rs ):
	print "row.count %d" % ( rs.getRowCount() )

	columnCount = rs.getColumnCount()

	hdrs = ""

	for i in range( columnCount ):
		hdrs = hdrs + ( "\t%s(%s)" % ( rs.getName( i ), rs.getType( i ) ) )

	print hdrs

	for i in range( rs.getRowCount() ):
		rs.moveTo( i )

		row = ""

		for j in range( columnCount ):
			row = row + ( "\t[%s]" % ( rs.getString( j ) ) )

		print row

if __name__ == "__main__":
	if len( sys.argv ) != 2:
		print "Usage: %s <config file>" % ( sys.argv[0] )
		print "\tpython %s ../../sphivedb/sphivedbcli.ini" % ( sys.argv[0] )
		sys.exit( -1 )

	configFile = sys.argv[1]

	cli = sphivedbcli.SPHiveDBClient()

	cli.init( configFile )

	try:
		resp = cli.execute( 0, "foobar", "addrbook", \
			[ "insert into addrbook ( addr ) values ( \"%d\" )" % ( time.time() ), \
			"select * from addrbook" ] )

		if 0 == resp.getErrorCode():
			rsCount = resp.getResultCount()

			for i in range( rsCount ):
				rs = resp.getResultSet( i )
				printResultSet( rs )
		else:
			print "%d: %s" % ( resp.getErrdataCode(), resp.getErrdataMsg() )

	except Exception, e:
		print e

File: client/php/testcli.php
<?php
/*
 * Copyright 2009 Stephen Liu
 * For license terms, see the file COPYING along with this library.
 */

include_once "sphivedbcli.php";

function printResultSet( $rs ) {
	printf( "row.count %d\n", $rs->getRowCount() );

	$columnCount = $rs->getColumnCount();

	$hdrs = "";
	for( $i = 0; $i < $columnCount; $i++ ) {
		$hdrs = $hdrs . "\t" . $rs->getName( $i ) . "(" . $rs->getType( $i ) . ")";
	}

	printf( "%s\n", $hdrs );

	$rowCount = $rs->getRowCount();

	for( $i = 0; $i < $rowCount; $i++ ) {
		$rs->moveTo( $i );

		$row = "";

		for( $j = 0; $j < $columnCount; $j++ ) {
			$row = $row . "\t[" . $rs->getField( $j ) . "]";
		}

		printf( "%s\n", $row );
	}
}

if( $argc != 2 ) {
	printf( "Usage: %s <config file>\n", $argv[0] );
	printf( "\tphp %s ../../sphivedb/sphivedbcli.ini\n", $argv[0] );
	return -1;
}

$configFile = $argv[1];

$client = new SPHiveDBClient();

$ret = $client->init( $configFile );

$respObj = $client->execute( 0, "foobar", "addrbook",
		array( "insert into addrbook ( addr ) values ( \"" . time() . "\" )",
			"select * from addrbook" ) );

if( 0 == $respObj->getErrorCode() ) {
	$rsCount = $respObj->getResultCount();

	for( $i = 0; $i < $rsCount; $i++ ) {
		$rs = $respObj->getResultSet( $i );
		printResultSet( $rs );
	}
} else {
	echo $respObj->getErrdataCode() . ": " . $respObj->getErrdataMsg();
}

?>

